File: app/src/main/java/com/yao/zhihudaily/model/RecommendsJson.java
package com.yao.zhihudaily.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by Administrator on 2016/9/16.
 */
public class RecommendsJson {

    private ArrayList<Item> items;
    @SerializedName("item_count")
    private int itemCount;


    public class Item {
        public int index;
        public ArrayList<Recommender> recommenders;
        public Author author;

        public int getIndex() {
            return index;
        }

        public void setIndex(int index) {
            this.index = index;
        }

        public ArrayList<Recommender> getRecommenders() {
            return recommenders;
        }

        public void setRecommenders(ArrayList<Recommender> recommenders) {
            this.recommenders = recommenders;
        }

        public Author getAuthor() {
            return author;
        }

        public void setAuthor(Author author) {
            this.author = author;
        }

        @Override
        public String toString() {
            return "Item{" +
                    "index=" + index +
                    ", recommenders=" + recommenders +
                    ", author=" + author +
                    '}';
        }
    }

    public class Author {
        public String name;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        @Override
        public String toString() {
            return "Author{" +
                    "name='" + name + '\'' +
                    '}';
        }
    }

    public ArrayList<Item> getItems() {
        return items;
    }

    public void setItems(ArrayList<Item> items) {
        this.items = items;
    }

    public int getItemCount() {
        return itemCount;
    }

    public void setItemCount(int itemCount) {
        this.itemCount = itemCount;
    }


    @Override
    public String toString() {
        return "RecommendsJson{" +
                "items=" + items +
                ", itemCount=" + itemCount +
                '}';
    }
}

File: app/src/main/java/com/yao/zhihudaily/model/RecommendsJson.kt
package com.yao.zhihudaily.model

import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * @author Yao
 * @date 2016/9/16
 */
class RecommendsJson {

    var items: ArrayList<Item>? = null
    @SerializedName("item_count")
    var itemCount: Int = 0


    inner class Item {
        var index: Int = 0
        lateinit var recommenders: ArrayList<Recommender>
        lateinit var author: Author

        override fun toString(): String {
            return "Item{" +
                    "index=" + index +
                    ", recommenders=" + recommenders +
                    ", author=" + author +
                    '}'.toString()
        }
    }

    inner class Author {
        lateinit var name: String

        override fun toString(): String {
            return "Author{" +
                    "name='" + name + '\''.toString() +
                    '}'.toString()
        }
    }


    override fun toString(): String {
        return "RecommendsJson{" +
                "items=" + items +
                ", itemCount=" + itemCount +
                '}'.toString()
    }
}
